protoc-gen-pbandk/lib/src/commonMain/kotlin/pbandk/gen/FileBuilder.kt

```diff
@@ -239,11 +239,43 @@ internal open class FileBuilder(val namer: Namer = Namer.Standard, val supportMa
             ?.associate { it.substringBefore("->") to it.substringAfter("->", "") }
             ?: emptyMap()
 
+        // Convert each mapping to a regex pattern, and then associate it with the replacement string
+        val packageMappingRegexes = packageMappings
+            .filter { it.value.contains("*") || it.value.contains("?") }
+            .map { (from, to) ->
+                from.replace(".", "\\.")
+                    .replace("*", "(.*)")
+                    .toRegex() to to
+            }
+
+        private fun matchPackageNameWithRegex(): String? {
+            val packageName = fileDesc.`package` ?: return null
+
+            if (packageMappings.containsKey(packageName))
+                return packageMappings[packageName]
+
+            val firstRegexMatch = packageMappingRegexes.firstNotNullOfOrNull { (from, to) ->
+                from.matchEntire(packageName)?.let { it to to }
+            }
+
+            return firstRegexMatch?.let { (from, to) ->
+                val splitTo = to.split("*")
+                val groups = from.groupValues
+
+                // If there is only one element, reduce will fail.
+                if (splitTo.size == 1)
+                    return splitTo[0] + groups[1]
+
+                // Replace each * in to string with regex matches in order.
```

Comment:
Do we need the complexity of supporting multiple wildcards in the pattern? Is that something you're using for your use case? If not, I think it'd be better to keep this simple and only allow a pattern with a single wildcard on the end.

Reply:
Makes sense, mostly we care about matching the prefix and just remapping. But I was also thinking about mapping some stuff like:
- `com.*.struct` -> `pbandk.*strukt` (assume auto-gen doesn't play with the package name `struct`). 
But I think it should be good enough for the first iteration. Future improvements can be done based on feedback.